independent-projects/resteasy-reactive/server/processor/src/main/java/org/jboss/resteasy/reactive/server/processor/scanning/CacheControlScanner.java

```diff
@@ -27,32 +29,55 @@ public class CacheControlScanner implements MethodScanner {
     @Override
     public List<HandlerChainCustomizer> scan(MethodInfo method, ClassInfo actualEndpointClass,
             Map<String, Object> methodContext) {
+
+        ClassInfo currentClassInfo = method.declaringClass();
+
+        // Check whether the actualEndpointClass is implementing an interface containing the method
+        // In this case, allow annotations on the implementation itself to take precedence over annotations on the
+        // interface.
+        if (!actualEndpointClass.equals(currentClassInfo) && Modifier.isInterface(currentClassInfo.flags())) {
+            MethodInfo actualMethod = actualEndpointClass.method(method.name(), method.parameterTypes().toArray(new Type[0]));
+
+            // first check bean implementation
+            List<HandlerChainCustomizer> customizers = scanImpl(actualMethod, actualEndpointClass, methodContext);
+            if (customizers.isEmpty()) {
+                // if no annotations where found on implementation, check the interface
+                customizers = scanImpl(method, currentClassInfo, methodContext);
+            }
+            return customizers;
+        } else {
+            return scanImpl(method, actualEndpointClass, methodContext);
+        }
+    }
+
+    private List<HandlerChainCustomizer> scanImpl(MethodInfo methodInfo, ClassInfo classInfo,
```

Comment:
Can we rename this to `scanImpl`?

Reply:
I don't quite understand your comment. I have
- moved the original implementation of scan() to a new private method scanImpl() and renamed the parameters to something more general, because scanImpl is called with different arguments now
- changed the implementation of scan() to recognize the "bean-implements-interface" case and then call scanImpl once with the method&class of the implementation and then with the method&class of the interface
- in the else case, scanImpl is called with the original arguments, so it behaves exactly like the original implementation of scan()

What exactly should we rename?

Reply:
Oops, sorry I misstyped the comment. I meant rename `scanImpl` to `doScan`

Reply:
Done, commits squashed and rebased to current main.